src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import classNames from 'classnames';
+
+interface Props {
+  people: Person[];
+  debounceDelay?: number;
+  onSelected: (person: Person | null) => void;
+}
+
+// eslint-disable-next-line react/display-name
+const Autocomplete: React.FC<Props> = React.memo(
+  ({ people, debounceDelay = 300, onSelected }) => {
+    const [query, setQuery] = useState('');
+    const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+    const [debounceQuery, setdebounceQuery] = useState(query);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+      setSelectedPerson(null);
+    };
+
+    useEffect(() => {
+      const handler = setTimeout(() => setdebounceQuery(query), debounceDelay);
+
+      return () => clearTimeout(handler);
+    }, [query, debounceDelay]);
+
+    const filterePeople = useMemo(() => {
```

Comment:
There is a typo in the variable name `filterePeople`. It should be corrected to `filteredPeople` for clarity and consistency.